eth/consensus/clique/clique.py

```diff
@@ -0,0 +1,171 @@
+import logging
+import time
+from typing import Sequence, Iterable
+
+from eth.abc import (
+    AtomicDatabaseAPI,
+    BlockHeaderAPI,
+    VirtualMachineAPI,
+)
+from eth.db.chain import ChainDB
+
+from eth_typing import (
+    Address,
+    Hash32,
+)
+from eth_utils import (
+    encode_hex,
+    to_tuple,
+    ValidationError,
+)
+
+from eth.typing import (
+    HeaderParams,
+    VMConfiguration,
+    VMFork,
+)
+from eth.vm.chain_context import ChainContext
+from eth.vm.execution_context import (
+    ExecutionContext,
+)
+
+from .datatypes import (
+    Snapshot,
+)
+from .header_cache import HeaderCache
+from .snapshot_manager import SnapshotManager
+from ._utils import (
+    get_block_signer,
+    is_in_turn,
+    validate_header_integrity,
+)
+
+
+def create_execution_context(header: BlockHeaderAPI,
+                             prev_hashes: Iterable[Hash32],
+                             chain_context: ChainContext,
+                             import_block: BlockHeaderAPI=None) -> ExecutionContext:
+
+    # In Clique consensus, the tx fee goes to the signer
+    fee_recipient = None
+    if import_block is not None:
+        try:
+            fee_recipient = get_block_signer(import_block.header)
+        except Exception:
+            pass
+
+    return ExecutionContext(
+        coinbase=header.coinbase,
```

Comment:
Oh wait, I just found this in the spec.

![image](https://user-images.githubusercontent.com/521109/67860218-955c7a80-fb1d-11e9-9d46-2844fde60ba0.png)

It seems that Clique explicitly says that we should set the `signer` as the `coinbase` to also return the `signer` for the `COINBASE` opcode.

Do we still want to introduce `fee_recipient` to allow more fine grained control nonetheless?

Reply:
Cool :+1: